### PR TITLE
FEATURE: Add creation and modification timestamps to nodes

### PR DESCRIPTION
**Pull request by bwaidelich**
**Note:** This requires a migration that can be triggered via `./flow cr:setup`

Resolves: #4092

### CODE REVIEW

**Comment by bwaidelich** [comment]
Draft because:

- [x] Clarify which events update the timestamps (e.g. what timestamps will a specialized node get initially?)
- [x] Fix existing and add new tests
- ~[ ] Fix PostgreSQL support~ (out of scope)
- [x] Documentation

File: Neos.ContentRepository.BehavioralTests/Tests/Behavior/Features/NodePublishing/PublishMovedNodesWithoutDimensions.feature [hunk 1]
```diff
@@ -82,7 +82,7 @@ Feature: Publishing moved nodes without dimensions
       | nodesToPublish           | [{"contentStreamId": "user-cs-identifier", "dimensionSpacePoint": {}, "nodeAggregateId": "sir-david-nodenborough"}] |
     And the graph projection is fully up to date
 
-    Then I expect the graph projection to consist of exactly 4 nodes
+    Then I expect the graph projection to consist of exactly 6 nodes
```
Comment (bwaidelich) [inline Neos.ContentRepository.BehavioralTests/Tests/Behavior/Features/NodePublishing/PublishMovedNodesWithoutDimensions.feature]: I had to adjust the expected number of nodes here and in 3 cases below because moving a node now might "materialize" it in order to update the `lastModified` timestamp per affected content dimension

Comment (bwaidelich) [reply Neos.ContentRepository.BehavioralTests/Tests/Behavior/Features/NodePublishing/PublishMovedNodesWithoutDimensions.feature]: We agreed that node moving (and en/disabling) does not update the last modified timestamp, so this is no longer an issue

File: Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/Feature/NodeMove.php [hunk 2]
```diff
@@ -41,6 +42,9 @@ abstract protected function getTableNamePrefix(): string;
     private function whenNodeAggregateWasMoved(NodeAggregateWasMoved $event): void
     {
         $this->transactional(function () use ($event) {
+
+            $dimensionSpacePoints = [];
+
```
Comment (skurfuerst) [inline Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/Feature/NodeMove.php]: why is this needed? you never read from it

File: Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/NodeRecord.php [hunk 3]
```diff
@@ -107,7 +125,22 @@ public static function fromDatabaseRow(array $databaseRow): self
             SerializedPropertyValues::fromArray(json_decode($databaseRow['properties'], true)),
             NodeTypeName::fromString($databaseRow['nodetypename']),
             NodeAggregateClassification::from($databaseRow['classification']),
-            isset($databaseRow['name']) ? NodeName::fromString($databaseRow['name']) : null
+            isset($databaseRow['name']) ? NodeName::fromString($databaseRow['name']) : null,
+            Timestamps::create(
+                self::parseDateTimeString($databaseRow['created']),
+                self::parseDateTimeString($databaseRow['originalcreated']),
```
Comment (skurfuerst) [inline Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/NodeRecord.php:131]: this makes the change breaking, doesn't it? (in an existing instance). I'd like to discuss whether we can make it non breaking :)

File: Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Repository/ProjectionContentGraph.php [hunk 4]
```diff
@@ -177,9 +177,38 @@ public function getAnchorPointForNodeAndOriginDimensionSpacePointAndContentStrea
...
-        } else {
-            return null;
         }
+        return null;
+    }
+
+    /**
+     * @return iterable<NodeRelationAnchorPoint>
+     */
+    public function getAnchorPointsForNodeAndDimensionSpacePointsAndContentStream(
```
Comment (skurfuerst) [inline Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Repository/ProjectionContentGraph.php]: what is this new method doing? I didn't find where it is used

**Review by skurfuerst — DISMISSED** [review]
really nice, just 2-3 discussions / nitpicks then we can merge this IMHO

File: Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/Feature/NodeMove.php (hunk 2 above)
Comment (bwaidelich) [reply Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/Feature/NodeMove.php]: probably a left-over, I'll check

File: Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/NodeRecord.php (hunk 3 above)
Comment (bwaidelich) [reply Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/NodeRecord.php:131]: breaking because it requires a `./flow cr:setup` or why do you mean? Ah the projection replay.. Mh, yes, lets talk about it!

File: Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Repository/ProjectionContentGraph.php (hunk 4 above)
Comment (bwaidelich) [reply Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Repository/ProjectionContentGraph.php]: probably a left-over, too

File: Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/NodeRecord.php (hunk 3 above)
Comment (skurfuerst) [reply Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/NodeRecord.php:131]: breaking, if created is null - exactly :)

Comment (bwaidelich) [reply Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/NodeRecord.php:131]: FYI: `./flow cr:setup` will create the new columns with `0000-00-00 00:00:00` for the creation and `NULL`  for the modification values so that is not breaking (but it leads to wrong timestamps of course).
After `./flow cr:replay graph` everything looks good!
